packages/@react-spectrum/actiongroup/src/ActionGroup.tsx

```diff
@@ -308,6 +308,9 @@ function ActionGroupItem<T>({item, state, isDisabled, isEmphasized, staticColor,
             text: {
               id: hideButtonText ? textId : null,
               isHidden: hideButtonText
+            },
+            icon: {
+              UNSAFE_className: hideButtonText ? classNames(styles, 'spectrum-ActionGroup-item-icon') : null
```

Comment:
the conditional is needed because the Icon has some padding already applied for the icon only ActionButton case:
<img width="568" alt="image" src="https://github.com/adobe/react-spectrum/assets/9661127/b690d1ee-b0d9-485d-b804-210895f7cef4">
which we don't want to override with `spectrum-ActionGroup-itemIcon` unless our `spectrum-ActionGroup-item--iconOnly` is handling applying this padding already

Here are the cases:
1. We are in a default ActionGroup case. `spectrum-ActionGroup-item--iconOnly` doesn't get applied anywhere because we aren't in the "buttonLabelBehavior=hide" case. All ActionButtons within the ActionGroup have ActionButton specific classes that already handle creating the padding on the left/right to make the button properly sized, regardless if the ActionButton with or without accompanying text. We don't want to apply our `spectrum-ActionGroup-itemIcon' class either since everything is handled already.
 
2. We are in a ActionGroup with "buttonLabelBehavior=hide" case. `spectrum-ActionGroup-item--iconOnly` is applied, handling the `padding-inline-end` for icon only and icon + text ActionButtons. Since we are handling it there, we need to override the ActionButton's icon only styling shown at the very top so we don't double dip on `padding-inline-end`

I'll go ahead and change the className

Reply:
ooo, and useHasChild doesn't update because we're not removing the text, just hiding it
that's annoying